components/App.js

```diff
@@ -1,42 +1,40 @@
-import Inventory from './Inventory.js';
+import Menu from './Menu.js';
+import WeaponWheel from './WeaponWheel.js';
 import inventory from '../inventory.js';
-import {loginManager} from '../login.js';
-import {getState, setState, getSpecificState} from '../state.js';
-import {setBindings} from './bindings.js';
+import { loginManager } from '../login.js';
+import { state, getSpecificState } from '../state.js';
+import { setBindings } from './bindings.js';
 
-let appState = getState();
-
-let appProps = {
-    inventoryItems: appState.inventory.items,
-    selectedWeapon: appState.selectedWeapon,
-    isXR: appState.isXR
-};
+let appState = state;
 
 const onclickBindings = {
-  'inventory-wear': e => {
+  'inventory-wear': (e) => {
     const id = parseInt(e.target.getAttribute('inventoryid'), 10);
     loginManager.setAvatar(id);
   },
-  'inventory-discard': async e => {
+  'inventory-discard': async (e) => {
     const id = parseInt(e.target.getAttribute('inventoryid'), 10);
     await inventory.discardFile(id);
   },
-  'inventory-upload': e => {
+  'inventory-upload': (e) => {
     const file = document.getElementById("twoD-inventoryUploadBtn").files[0];
     inventory.uploadFile(file);
   },
 };
+
 inventory.addEventListener('filesupdate', (e) => {
-    updateProps({
-        inventoryItems: e.data,
-    })
+    state.menu.inventory.items = state.menu.inventory.items.concat(e.data)
```

Comment:
This re-introduces corruption in the inventory, which was already fixed once.

Getting a new files list here does not mean we are appending, it means we are replacing.